scheduler/src/cook/tools.clj

```diff
@@ -951,6 +952,39 @@
                 [usage' (below-quota? quota usage')]))]
       (filter-sequential filter-with-quota usage queue))))
 
+
+; This is used by the /unscheduled_jobs code to determine whether
+; or not to report rate-limiting as a reason for being pending
+(defonce pool->user->num-rate-limited-jobs (atom {}))
```

Comment:
How is this used?

Reply:
Ah nice, can we assert on the `/unscheduled_jobs` reason in the integration test?

Reply:
The test would be nondeterministic. After a rank cycle, we clear this variable. We update it during a match cycle, so in the gap between the match rank and match cycle, nothing is ever rate limited. (enforced by the laziness we lost in the other code review change).

Reply:
I wrote the test. Only way to see if its reliable is to write it and see if its reliable.